sdk/keyvault/azure-security-keyvault-administration/src/test/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClientTest.java

```diff
@@ -68,36 +66,37 @@ public void listRoleDefinitions(HttpClient httpClient) {
     @ParameterizedTest(name = DISPLAY_NAME)
     @MethodSource("com.azure.security.keyvault.administration.KeyVaultAdministrationClientTestBase#createHttpClients")
     public void setRoleDefinition(HttpClient httpClient) {
-        if (getTestMode() != TestMode.PLAYBACK) {
-            // Currently there is no test cloud environment for Managed HSM.
-            // TODO: Remove once there is a proper cloud environment available.
-            return;
-        }
+        Assumptions.assumeTrue(isHsmDeployed);
 
         asyncClient = getClientBuilder(httpClient, false).buildAsyncClient();
+        String roleDefinitionName = (getTestMode() == TestMode.PLAYBACK || getTestMode() == TestMode.RECORD)
```

Comment:
On live tests we use a random UUID for creating `roleDefinitionName`s and `roleAssignmentName`s. If we record using a random UUID, during playback the framework will complain that the URL in the request is different.

Reply:
Mind filing an issue for this? Also, could you try using https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core-test/src/main/java/com/azure/core/test/utils/TestResourceNamer.java#L84 and seeing if this would work

Reply:
FWIW, in .NET we have a helper that creates random numbers (including a GUID, i.e. 128-bit number) based on a seed stored in the recording that should never change. In live mode, it just returns actual random numbers or, in the case of GUIDs, a newly generated GUID from our `Guid.NewGuid()` base class library method.